ui/src/shared/components/StaticLegend.js

```diff
@@ -0,0 +1,120 @@
+import React, {PropTypes, Component} from 'react'
+import _ from 'lodash'
+import uuid from 'node-uuid'
+
+const style = {
+  position: 'absolute',
+  width: 'calc(100% - 32px)',
+  bottom: '8px',
+  left: '16px',
+  paddingTop: '8px',
+}
+
+const removeMeasurement = (label = '') => {
+  const [measurement] = label.match(/^(.*)[.]/g) || ['']
+  return label.replace(measurement, '')
+}
+
+const staticLegendItemClassname = (visibilities, i, hoverEnabled) => {
+  if (visibilities.length) {
+    return `${hoverEnabled
+      ? 'static-legend--item'
+      : 'static-legend--single'}${visibilities[i] ? '' : ' disabled'}`
+  }
+
+  // all series are visible to match expected initial state
+  return 'static-legend--item'
+}
+
+class StaticLegend extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      visibilities: [],
+    }
+  }
+
+  componentDidMount = () => {
+    const {height} = this.staticLegendRef.getBoundingClientRect()
+    this.props.handleReceiveStaticLegendHeight(height)
+  }
+
+  componentDidUpdate = () => {
+    const {height} = this.staticLegendRef.getBoundingClientRect()
+    this.props.handleReceiveStaticLegendHeight(height)
+  }
+
+  componentWillUnmount = () => {
+    this.props.handleReceiveStaticLegendHeight(null)
+  }
+
+  handleClick = i => e => {
+    const visibilities = this.props.dygraph.visibility()
+
+    if (e.shiftKey || e.metaKey) {
+      visibilities[i] = !visibilities[i]
+      this.props.dygraph.setVisibility(visibilities)
+      this.setState({visibilities})
+      return
+    }
+
+    const newVisibilities = visibilities[i]
+      ? _.map(visibilities, v => !v)
+      : _.map(visibilities, () => false)
+    newVisibilities[i] = true
```

Comment:
- If this item is already hidden, set everything to false, and set pertinent visibility to true.
- If this item is already visible, toggle all visibilities, and set the pertinent visibility to true.

Am I reading the second condition correctly? What happens if you shift-click a bunch of measurements, and half are on and half are off?

Reply:
you are correct....... fixing